app/src/main/java/com/mapbox/services/android/navigation/testapp/activity/navigationui/NavigationLauncherActivity.java

```diff
@@ -96,6 +107,32 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
     mapView.onCreate(savedInstanceState);
     mapView.getMapAsync(this);
     localeUtils = new LocaleUtils();
+    ReplayRouteDto routeFromJson = new GsonBuilder().create().fromJson(obtainJson("reroute.json"),
+      ReplayRouteDto.class);
+    ReplayLocationMapper mapper = new ReplayLocationMapper(routeFromJson.getLocations());
+    replayLocationDispatcher = new ReplayLocationDispatcher(mapper.toLocations());
+    replayLocationDispatcher.addReplayLocationListener(location -> {
+      String dateFormatPattern = "yyyy-MM-dd'T'HH:mm:ss.SSSZ";
+      SimpleDateFormat dateFormat = new SimpleDateFormat(dateFormatPattern);
+      dateFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
+      String parsedTime = dateFormat.format(location.getTime());
+      Timber.d("parsedTime %s", parsedTime);
+    });
```

Comment:
Extracting this `Replay` setup into an `initializeReplayLocationDispatcher` method would help clean up `onCreate`

Reply:
> ☝️ was the approach taken for the GPX classes. I guess, we could have a ReplayLocationEngine publicly available and the rest of the classes package-private. As commented #1039 (comment)

Yep, was going to ask about this. thanks!